Microsoft.Toolkit.Uwp.UI.Media/Extensions/System.Threading.Tasks/AsyncMutex.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Toolkit.Uwp.UI.Media.Extensions.System.Threading.Tasks
+{
+    /// <summary>
+    /// An <see langword="async"/> <see cref="AsyncMutex"/> implementation that can be easily used inside a <see langword="using"/> block
```

Comment:
FYI, I usually use [Nito.AsyncEx](https://www.nuget.org/packages/Nito.AsyncEx) for this purpose. Though for the one use case here it doesn't make sense to pull in the dependency on it.  We should just make sure we leave it as internal.

Reply:
Ah, yeah I'm aware of that lib though in this case I figured there was no point in getting another dependency just for this, as you said. Plus I believe this implementation is more lightweight than the one in `Nito.AsyncEx`, from what I've seen. And yup, this should definitely remain internal 👍